packages/backend-deployer/src/cdk_deployer_singleton_factory.ts

```diff
@@ -55,18 +57,37 @@ export class BackendDeployerFactory {
    */
   constructor(
     private readonly packageManagerController: PackageManagerController,
-    private readonly formatter: BackendDeployerOutputFormatter
+    private readonly formatter: BackendDeployerOutputFormatter,
+    private readonly backendDeployerIOHost: AmplifyIOHost
   ) {}
 
   /**
    * Returns a single instance of BackendDeployer
    */
   getInstance(): BackendDeployer {
     if (!BackendDeployerFactory.instance) {
+      let profile = undefined;
+      if (process && process.argv) {
+        for (let i = 2; i < process.argv.length; i++) {
+          if (process.argv[i] == '--profile') {
+            profile = process.argv[i + 1];
+          }
+        }
+      }
```

Comment:
this is temporary ?

We should `getInstance(props? : {profile?:string})`

Reply:
I'm having difficulty patching this correctly. `profile` is an input to the `CdkToolkit` constructor. However `profile` is only available at command runtime while we are injecting/creating `CdkToolkit` in the factories instantiation.

Reply:
We can have logic like this, we do this here for example https://github.com/aws-amplify/amplify-backend/blob/32bcc0bde671937151b078f3e6cb58d608642a26/packages/cli-core/src/printer.ts#L3-L5 .

But this should move to cli package I would say, the argv access.

Reply:
yeah that was the inspiration. I'll try to bring this value from cli.

Reply:
I moved this to be passed in the factory constructor as it's not really an instance variable, i.e. we are not going to have multiple instances per profile.

Reply:
A side effect of this change is an API change to `cdk_deployer` that it can't accept `profile` anymore as a runtime argument. Now the question is if it's considered a public facing breaking change for us to do major version release or move instantiation of `CDKToolkit` inside `deploy/destroy` methods.

Reply:
But... It might be impossible to honor method level override with this new design. So let's do MV.

Reply:
Updated.